### PR TITLE
Extend the monitor notification interface with endpoint id getter

### PR DESCRIPTION
**Pull request by aditighag**
See commit message.

### CODE REVIEW

**Review by joestringer — APPROVED** [review]
I didn't know about the nil cast thing for compile time assertion on types. Nice :)

File: pkg/hubble/parser/agent/parser_test.go [hunk 1]
```diff
@@ -45,6 +45,7 @@ func (e *mockEndpoint) GetID() uint64           { return e.ID }
 func (e *mockEndpoint) GetOpLabels() []string   { return e.Labels }
 func (e *mockEndpoint) GetK8sPodName() string   { return e.PodName }
 func (e *mockEndpoint) GetK8sNamespace() string { return e.Namespace }
+func (e *mockEndpoint) GetID16() uint16         { return uint16(e.ID) }
```
Comment (rolinh) [inline pkg/hubble/parser/agent/parser_test.go]: Could we instead update `mockEndpoint.ID` to be `uint16` and cast to `uint64` for `GetID()`?

Comment (aditighag) [reply pkg/hubble/parser/agent/parser_test.go]: Done

**Comment by aditighag** [comment]
test-me-please

**Review by christarazi — APPROVED** [review]

**Review by rolinh — APPROVED** [review]

**Comment by aditighag** [comment]
test-1.20-4.9

**Comment by aditighag** [comment]
I see that this test "Checks ClusterIP Connectivity Checks DualStack services" has been failing on other PRs too - 

https://jenkins.cilium.io/job/Cilium-PR-K8s-1.20-kernel-4.9/1012/testReport/Suite-k8s-1/20/K8sServicesTest_Checks_ClusterIP_Connectivity_Checks_DualStack_services/history/

**Comment by aditighag** [comment]
test-1.20-4.9

**Comment by aditighag** [comment]
test-1.20-4.9

**Comment by aditighag** [comment]
One reason why I feel the test failures are unrelated because this PR only extends an existing interface. If there were issues related to run time cast due to which the interface isn't being satisfied, then we would've seen more obvious and deterministic failures.

Since all the other tests had passed in earlier runs, I'm triggering only  the`test-1.20-4.9` test.

**Comment by aditighag** [comment]
test-1.20-4.9

**Comment by aditighag** [comment]
Quay is down. Will need to re-try.

**Comment by aditighag** [comment]
test-1.20-4.9

**Comment by aditighag** [comment]
>08:05:27  VBoxManage: error: Details: code VBOX_E_INVALID_OBJECT_STATE (0x80bb0007), component MediumWrap, interface IMedium
08:05:27  VBoxManage: error: Context: "RTEXITCODE handleUnregisterVM(HandlerArg*)" at line 162 of file VBoxManageMisc.cpp

Provisioning related issue.

**Pull request merged**